clientdb/order.go

```diff
@@ -747,10 +777,19 @@ func deserializeOrderTlvData(r io.Reader, o order.Order) error {
 // serializeOrderTlvData encodes all additional data of an order as a single tlv
 // stream.
 func serializeOrderTlvData(w io.Writer, o order.Order) error {
-	var tlvRecords []tlv.Record
+	var (
+		tlvRecords                 []tlv.Record
+		askAnnouncementConstraints uint8
+		bidUnannoucedChannels      bool
+	)
 
 	switch castOrder := o.(type) {
 	case *order.Ask:
+		noPreference := order.AnnouncementNoPreference
+		if castOrder.AnnouncementConstraints != noPreference {
```

Comment:
Do we really need this check when we have `if askAnnouncementConstraints != 0` further down?

Reply:
TLVs types need to be added in increasing order. Is there any "more clear way" to do it?

Reply:
I mean we could just set `askAnnouncementConstraints = uint8(castOrder.AnnouncementConstraints))` here without the `if castOrder.AnnouncementConstraints != noPreference {` as we check for `if askAnnouncementConstraints != 0 {` again further down.